docs/weave-tutorial/09-application-level-codec.md

```diff
@@ -0,0 +1,49 @@
+---
+id: app-level-codec
+title: Application Level Codec
+sidebar_label: Application Level Codec
+---
+
+We created a `codec` in [Codec documentation](weave-tutorial/codec), only thing left for the codec is wrapping it up with a generalizable transaction format. With this we will define a standart way of communications in Weave based chains. Application level codec is placed at `cmd/blog` where the application wraps up.
+
+```protobuf
+message Tx {
+  // Enables coin.GetFees()
+  cash.FeeInfo fees = 1;
+  //StdSignature represents the signature, the identity of the signer
+  // (the Pubkey), and a sequence number to prevent replay attacks.
+  repeated sigs.StdSignature signatures = 2;
+  // ID of a multisig contract.
+  repeated bytes multisig = 4;
+  // sum defines over all allowed messages on this chain.
+  oneof sum {
+    cash.SendMsg cash_send_msg = 51;
+    multisig.CreateMsg multisig_create_msg = 56;
+    multisig.UpdateMsg multisig_update_msg = 57;
+    validators.ApplyDiffMsg validators_apply_diff_msg = 58;
+    ExecuteBatchMsg execute_batch_msg = 60;
+    migration.UpgradeSchemaMsg migration_upgrade_schema_msg = 69;
+
+    blog.CreateUserMsg blog_create_user_msg = 100;
+    blog.CreateBlogMsg blog_create_blog_msg = 101;
+    blog.ChangeBlogOwnerMsg blog_change_blog_owner_msg = 102;
+    blog.CreateArticleMsg blog_create_article_msg = 103;
+    blog.DeleteArticleMsg blog_delete_article_msg = 104;
+    blog.CancelDeleteArticleTaskMsg blog_cancel_delete_article_task_msg = 105;
+    blog.CreateCommentMsg blog_create_comment_msg = 106;
+    blog.CreateLikeMsg blog_create_like_msg = 107;
+  }
+}
+```
+
+**Tx** contains the message the general application message. It cointains fee info, signatures, multisig, and one of the module messages. When extending the **Tx**, follow the rules:
```

Comment:
`It looks like there is a wrd missing in "**Tx** contains the message the general application message. " also.`